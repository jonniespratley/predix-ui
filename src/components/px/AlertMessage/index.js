import React from 'react';
import styled, { css, keyframes } from 'styled-components';
import PropTypes from 'prop-types';
import Button from '../Button';

const fadeIn = keyframes`
  from {
    opacity: 0;
    left: 4rem;
  }
  to {
    opacity: 1;
    left: 0;
  }
`;
const fadeOut = keyframes`
  from {
    opacity: 1;
    max-height: 6.66667rem;
    min-height: 0px;
  }
  to {
    opacity: 0;
    max-height: 0px;
    min-height: 0px;
    padding-top: 0px;
    padding-bottom: 0px;
  }
`;

const AlertMessage = styled.div`
  visibility: hidden;
  display: none;
  width: var(--px-alert-message-width, auto);
  margin: 0.33333rem;
  padding: 0.66667rem;
  min-height: 4rem;
  word-wrap: break-word;
  word-break: break-word;
  background-color: var(--px-alert-message-background-color, white);
  position: relative;
  min-height: 0px;
  opacity: 0;
  max-height: 0px;
  padding-top: 0px;
  padding-bottom: 0px;
  box-shadow: 0px 4px 8px var(--px-shadow-notification, rgba(0, 0, 0, 0.2));
  animation: ${fadeOut} 0.4s cubic-bezier(0.78, 0.13, 0.16, 0.87);
  align-items: center;

  ${props => props.visible && css`
    overflow:hidden;
    height: 4rem;
    visibility: visible;
    display: flex;
    opacity: 1;
    max-height: none;
    padding-top: 0.66667rem;
    padding-bottom: 0.66667rem;
    animation: ${fadeIn} 0.4s cubic-bezier(0.78, 0.13, 0.16, 0.87);
  `}
  button {
    color: var(--px-alert-message-dismiss-icon-color, gray);
    outline: none;
    max-width: 50px;

    svg {
      height: 1.33333rem;
      width: 1.33333rem;
      fill: var(--px-alert-message-dismiss-icon-color, gray);
      stroke: var(--px-alert-message-dismiss-icon-color, gray);
    }
  }
`;

const AlertMessageContainer = styled.div`
  display: flex;
  align-items: center;
  flex: 2;
  margin-left: 0.66667rem;
  margin-right: 0.66667rem;
`;

const AlertMessageActions = styled.div`
  display: flex;
  align-items: center;
  position: relative;
  .dismiss {
    width: 44px;
    height: 44px;
    min-width: auto;
  }
`;

const AlertMessageSeverity = styled.div`
  min-height: 4rem;
  width: 4rem;
  margin-left: -0.6666rem;
  margin-top: -0.6666rem;
  margin-bottom: -0.6666rem;
  padding-top: 1rem;

  ${props => props.type === 'important' && css`
    background-color: var(--px-alert-message-color--important, red);
  `}

  ${props => (props.type === 'information' || props.type === 'info') && css`
    background-color: var(--px-alert-message-color--information, blue);
  `}

  ${props => props.type === 'warning' && css`
    background-color: var(--px-alert-message-color--warning, orange);
  `}

  ${props => props.type === 'error' && css`
    background-color: var(--px-alert-message-color--error, yellow);
  `}

  ${props => props.type === 'unknown' && css`
    background-color: var(--px-alert-message-color--unknown, gray);
  `}

  ${props => props.type === 'healthy' && css`
    background-color: var(--px-alert-label-background-color--healthy,green);
  `}

  ${props => props.type === 'success' && css`
    background-color: var(--px-alert-label-background-color--healthy,green);
  `}
`;

const Message = styled.div`
  color: var(--px-alert-message-text-color, black);
  line-height: 1.06667rem;
  font-size: 0.8rem;
  padding: 0;
  margin: 0;
`;

/**
 * AlertMessage component
 */
const Component = ({
  messageTitle,
  message,
  action,
  visible,
  actions,
  onActionClick,
  onDismiss,
  type,
  children
}) => (
  <AlertMessage visible={visible}>
    <AlertMessageSeverity type={type} />
    <AlertMessageContainer>
      <Message>
        <strong>{messageTitle} </strong>
        <span>{message}</span>
        <div>{children}</div>
      </Message>
    </AlertMessageContainer>
    <AlertMessageActions>
      {!actions && action === 'dismiss' &&
        <Button onClick={onDismiss} theme="bare" className="dismiss">
          <svg viewBox="0 0 22 22" preserveAspectRatio="xMidYMid meet" focusable="false">
            <g>
              <path strokeMiterlimit="10" d="M3 19L19 3M3 3l16 16" />
            </g>
          </svg>
        </Button>}

      {actions && actions()}

      {!actions && action === 'acknowledge' &&
        <Button
          id="actionButton"
          onClick={onActionClick}
          theme="tertiary"
        >OK
        </Button>}
    </AlertMessageActions>
  </AlertMessage>
);

Component.defaultProps = {
  action: null,
  type: 'information',
  message: null,
  messageTitle: null,
  visible: false,
  actions: null,
  children: null,
  onActionClick: null,
  onDismiss: null
};

Component.propTypes = {
  action: PropTypes.string,
  type: PropTypes.string,
  message: PropTypes.string,
  messageTitle: PropTypes.string,
  visible: PropTypes.bool,
  actions: PropTypes.func,
  onActionClick: PropTypes.func,
  onDismiss: PropTypes.func,
  children: PropTypes.node
};

Component.displayName = 'AlertMessage';

export default Component;
