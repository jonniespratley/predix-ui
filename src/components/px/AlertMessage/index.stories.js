import React from 'react';
import { storiesOf } from '@storybook/react';

//addons
import { action } from '@storybook/addon-actions';
import { withInfo } from '@storybook/addon-info';
import { withKnobs, text, array, object, boolean, number, select } from '@storybook/addon-knobs';

import AlertMessage from './';

const typeOptions = {
  information: 'Info',
  error: 'Error',
  important: 'Important',
  healthy: 'Healthy',
  warning: 'Warning',
  success: 'Success'
};
const navItems = [
  {
    "path": "/page1",
    "icon": "px-fea:analysis",
    "label": "Page 1"
  },
  {
    "path": "/page2",
    "icon": "px-fea:dashboard",
    "label": "Page 2"
  }
];
///
storiesOf('Alert Message', module)
  .addDecorator(withKnobs)
	.add('default', () => (
		<AlertMessage
      onDismissClick={action('onDismissClick')}
      onActionClick={action('onActionClick')}
      type={select('type', Object.keys(typeOptions))}
      action={select('action', ['acknowledge', 'dismiss'])}
      message={text('message', 'This is the message content.')}
      messageTitle={text('messageTitle', 'Alert')}
      visible={boolean('visible', true)}/>
  ))
  .add('with message', () => (
		<AlertMessage
      messageTitle='Alert'
      message='This is an information message'
      visible/>
  ))
  .add('with type "information"', () => (
		<AlertMessage
      messageTitle='Alert'
      message='This is a message'
      type='information'
      visible/>
  ))
  .add('with type "important"', () => (
		<AlertMessage
      messageTitle='Alert'
      message='This is a message'
      type='important'
      visible/>
  ))
  .add('with type "error"', () => (
		<AlertMessage
      messageTitle='Alert'
      message='This is a message'
      type='error'
      visible/>
  ))
  .add('with type "warning"', () => (
		<AlertMessage
      messageTitle='Alert'
      message='This is a message'
      type='warning'
      visible/>
  ))
  .add('with type "unkown"', () => (
		<AlertMessage
      messageTitle='Alert'
      message='This is a message'
      type='unkown'
      visible/>
	))
	;
