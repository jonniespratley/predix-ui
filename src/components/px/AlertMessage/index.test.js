//import { expect } from 'chai';
import React from 'react';
import { shallow } from 'enzyme';
import AlertMessage from './';
import renderer from 'react-test-renderer';

describe('AlertMessage', () => {
  test('should render', () =>{
    const wrapper = shallow(
      <AlertMessage/>
    );
  });

  test('matches snapshot', () => {
    const tree = renderer.create(<AlertMessage type='important' visible />).toJSON();
    expect(tree).toMatchSnapshot();
    expect(tree).toHaveStyleRule('display', 'flex');
    //expect(tree).toHaveStyleRule('color', 'var(--px-alert-label-text-color--information,white)');
  });
  //expect(wrapper.find('.label')).to.have.length(1);
  //expect(wrapper.find('.delta')).to.have.length(1);
  //expect(wrapper.find('.alpha')).to.have.length(1);
  //expect(wrapper.contains(<div className='label'/>)).to.equal(true);
});
