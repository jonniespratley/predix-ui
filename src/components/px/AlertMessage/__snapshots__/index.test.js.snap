// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`AlertMessage error matches snapshot 1`] = `
.c0 {
  visibility: hidden;
  display: none;
  width: var(--px-alert-message-width,auto);
  margin: 0.33333rem;
  padding: 0.66667rem;
  min-height: 4rem;
  word-wrap: break-word;
  word-break: break-word;
  background-color: var(--px-alert-message-background-color,white);
  position: relative;
  min-height: 0px;
  opacity: 0;
  max-height: 0px;
  padding-top: 0px;
  padding-bottom: 0px;
  box-shadow: 0px 4px 8px var(--px-shadow-notification,rgba(0,0,0,0.2));
  -webkit-animation: LkNfV 0.4s cubic-bezier(0.78,0.13,0.16,0.87);
  animation: LkNfV 0.4s cubic-bezier(0.78,0.13,0.16,0.87);
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
  overflow: hidden;
  height: 4rem;
  visibility: visible;
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  opacity: 1;
  max-height: none;
  padding-top: 0.66667rem;
  padding-bottom: 0.66667rem;
  -webkit-animation: ayxbO 0.4s cubic-bezier(0.78,0.13,0.16,0.87);
  animation: ayxbO 0.4s cubic-bezier(0.78,0.13,0.16,0.87);
}

.c0 button {
  color: var(--px-alert-message-dismiss-icon-color,gray);
  outline: none;
  max-width: 50px;
}

.c0 button svg {
  height: 1.33333rem;
  width: 1.33333rem;
  fill: var(--px-alert-message-dismiss-icon-color,gray);
  stroke: var(--px-alert-message-dismiss-icon-color,gray);
}

.c2 {
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
  -webkit-flex: 2;
  -ms-flex: 2;
  flex: 2;
  margin-left: 0.66667rem;
  margin-right: 0.66667rem;
}

.c4 {
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
  position: relative;
}

.c4 .dismiss {
  width: 44px;
  height: 44px;
  min-width: auto;
}

.c1 {
  min-height: 4rem;
  width: 4rem;
  margin-left: -0.6666rem;
  margin-top: -0.6666rem;
  margin-bottom: -0.6666rem;
  padding-top: 1rem;
  background-color: var(--px-alert-message-color--error,yellow);
}

.c3 {
  color: var(--px-alert-message-text-color,black);
  line-height: 1.06667rem;
  font-size: 0.8rem;
  padding: 0;
  margin: 0;
}

<div
  className="c0"
>
  <div
    className="c1"
    type="error"
  />
  <div
    className="c2"
  >
    <div
      className="c3"
    >
      <strong>
         
      </strong>
      <span />
      <div>
        This is a message
      </div>
    </div>
  </div>
  <div
    className="c4"
  />
</div>
`;

exports[`AlertMessage healthy matches snapshot 1`] = `
.c0 {
  visibility: hidden;
  display: none;
  width: var(--px-alert-message-width,auto);
  margin: 0.33333rem;
  padding: 0.66667rem;
  min-height: 4rem;
  word-wrap: break-word;
  word-break: break-word;
  background-color: var(--px-alert-message-background-color,white);
  position: relative;
  min-height: 0px;
  opacity: 0;
  max-height: 0px;
  padding-top: 0px;
  padding-bottom: 0px;
  box-shadow: 0px 4px 8px var(--px-shadow-notification,rgba(0,0,0,0.2));
  -webkit-animation: LkNfV 0.4s cubic-bezier(0.78,0.13,0.16,0.87);
  animation: LkNfV 0.4s cubic-bezier(0.78,0.13,0.16,0.87);
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
  overflow: hidden;
  height: 4rem;
  visibility: visible;
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  opacity: 1;
  max-height: none;
  padding-top: 0.66667rem;
  padding-bottom: 0.66667rem;
  -webkit-animation: ayxbO 0.4s cubic-bezier(0.78,0.13,0.16,0.87);
  animation: ayxbO 0.4s cubic-bezier(0.78,0.13,0.16,0.87);
}

.c0 button {
  color: var(--px-alert-message-dismiss-icon-color,gray);
  outline: none;
  max-width: 50px;
}

.c0 button svg {
  height: 1.33333rem;
  width: 1.33333rem;
  fill: var(--px-alert-message-dismiss-icon-color,gray);
  stroke: var(--px-alert-message-dismiss-icon-color,gray);
}

.c2 {
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
  -webkit-flex: 2;
  -ms-flex: 2;
  flex: 2;
  margin-left: 0.66667rem;
  margin-right: 0.66667rem;
}

.c4 {
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
  position: relative;
}

.c4 .dismiss {
  width: 44px;
  height: 44px;
  min-width: auto;
}

.c1 {
  min-height: 4rem;
  width: 4rem;
  margin-left: -0.6666rem;
  margin-top: -0.6666rem;
  margin-bottom: -0.6666rem;
  padding-top: 1rem;
  background-color: var(--px-alert-label-background-color--healthy,green);
}

.c3 {
  color: var(--px-alert-message-text-color,black);
  line-height: 1.06667rem;
  font-size: 0.8rem;
  padding: 0;
  margin: 0;
}

<div
  className="c0"
>
  <div
    className="c1"
    type="healthy"
  />
  <div
    className="c2"
  >
    <div
      className="c3"
    >
      <strong>
         
      </strong>
      <span />
      <div>
        This is a message
      </div>
    </div>
  </div>
  <div
    className="c4"
  />
</div>
`;

exports[`AlertMessage hidden - matches snapshot 1`] = `
.c0 {
  visibility: hidden;
  display: none;
  width: var(--px-alert-message-width,auto);
  margin: 0.33333rem;
  padding: 0.66667rem;
  min-height: 4rem;
  word-wrap: break-word;
  word-break: break-word;
  background-color: var(--px-alert-message-background-color,white);
  position: relative;
  min-height: 0px;
  opacity: 0;
  max-height: 0px;
  padding-top: 0px;
  padding-bottom: 0px;
  box-shadow: 0px 4px 8px var(--px-shadow-notification,rgba(0,0,0,0.2));
  -webkit-animation: LkNfV 0.4s cubic-bezier(0.78,0.13,0.16,0.87);
  animation: LkNfV 0.4s cubic-bezier(0.78,0.13,0.16,0.87);
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
}

.c0 button {
  color: var(--px-alert-message-dismiss-icon-color,gray);
  outline: none;
  max-width: 50px;
}

.c0 button svg {
  height: 1.33333rem;
  width: 1.33333rem;
  fill: var(--px-alert-message-dismiss-icon-color,gray);
  stroke: var(--px-alert-message-dismiss-icon-color,gray);
}

.c2 {
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
  -webkit-flex: 2;
  -ms-flex: 2;
  flex: 2;
  margin-left: 0.66667rem;
  margin-right: 0.66667rem;
}

.c4 {
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
  position: relative;
}

.c4 .dismiss {
  width: 44px;
  height: 44px;
  min-width: auto;
}

.c1 {
  min-height: 4rem;
  width: 4rem;
  margin-left: -0.6666rem;
  margin-top: -0.6666rem;
  margin-bottom: -0.6666rem;
  padding-top: 1rem;
  background-color: var(--px-alert-message-color--important,red);
}

.c3 {
  color: var(--px-alert-message-text-color,black);
  line-height: 1.06667rem;
  font-size: 0.8rem;
  padding: 0;
  margin: 0;
}

<div
  className="c0"
>
  <div
    className="c1"
    type="important"
  />
  <div
    className="c2"
  >
    <div
      className="c3"
    >
      <strong>
         
      </strong>
      <span />
      <div />
    </div>
  </div>
  <div
    className="c4"
  />
</div>
`;

exports[`AlertMessage important matches snapshot 1`] = `
.c0 {
  visibility: hidden;
  display: none;
  width: var(--px-alert-message-width,auto);
  margin: 0.33333rem;
  padding: 0.66667rem;
  min-height: 4rem;
  word-wrap: break-word;
  word-break: break-word;
  background-color: var(--px-alert-message-background-color,white);
  position: relative;
  min-height: 0px;
  opacity: 0;
  max-height: 0px;
  padding-top: 0px;
  padding-bottom: 0px;
  box-shadow: 0px 4px 8px var(--px-shadow-notification,rgba(0,0,0,0.2));
  -webkit-animation: LkNfV 0.4s cubic-bezier(0.78,0.13,0.16,0.87);
  animation: LkNfV 0.4s cubic-bezier(0.78,0.13,0.16,0.87);
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
  overflow: hidden;
  height: 4rem;
  visibility: visible;
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  opacity: 1;
  max-height: none;
  padding-top: 0.66667rem;
  padding-bottom: 0.66667rem;
  -webkit-animation: ayxbO 0.4s cubic-bezier(0.78,0.13,0.16,0.87);
  animation: ayxbO 0.4s cubic-bezier(0.78,0.13,0.16,0.87);
}

.c0 button {
  color: var(--px-alert-message-dismiss-icon-color,gray);
  outline: none;
  max-width: 50px;
}

.c0 button svg {
  height: 1.33333rem;
  width: 1.33333rem;
  fill: var(--px-alert-message-dismiss-icon-color,gray);
  stroke: var(--px-alert-message-dismiss-icon-color,gray);
}

.c2 {
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
  -webkit-flex: 2;
  -ms-flex: 2;
  flex: 2;
  margin-left: 0.66667rem;
  margin-right: 0.66667rem;
}

.c4 {
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
  position: relative;
}

.c4 .dismiss {
  width: 44px;
  height: 44px;
  min-width: auto;
}

.c1 {
  min-height: 4rem;
  width: 4rem;
  margin-left: -0.6666rem;
  margin-top: -0.6666rem;
  margin-bottom: -0.6666rem;
  padding-top: 1rem;
  background-color: var(--px-alert-message-color--important,red);
}

.c3 {
  color: var(--px-alert-message-text-color,black);
  line-height: 1.06667rem;
  font-size: 0.8rem;
  padding: 0;
  margin: 0;
}

<div
  className="c0"
>
  <div
    className="c1"
    type="important"
  />
  <div
    className="c2"
  >
    <div
      className="c3"
    >
      <strong>
         
      </strong>
      <span />
      <div>
        This is a message
      </div>
    </div>
  </div>
  <div
    className="c4"
  />
</div>
`;

exports[`AlertMessage information matches snapshot 1`] = `
.c0 {
  visibility: hidden;
  display: none;
  width: var(--px-alert-message-width,auto);
  margin: 0.33333rem;
  padding: 0.66667rem;
  min-height: 4rem;
  word-wrap: break-word;
  word-break: break-word;
  background-color: var(--px-alert-message-background-color,white);
  position: relative;
  min-height: 0px;
  opacity: 0;
  max-height: 0px;
  padding-top: 0px;
  padding-bottom: 0px;
  box-shadow: 0px 4px 8px var(--px-shadow-notification,rgba(0,0,0,0.2));
  -webkit-animation: LkNfV 0.4s cubic-bezier(0.78,0.13,0.16,0.87);
  animation: LkNfV 0.4s cubic-bezier(0.78,0.13,0.16,0.87);
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
  overflow: hidden;
  height: 4rem;
  visibility: visible;
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  opacity: 1;
  max-height: none;
  padding-top: 0.66667rem;
  padding-bottom: 0.66667rem;
  -webkit-animation: ayxbO 0.4s cubic-bezier(0.78,0.13,0.16,0.87);
  animation: ayxbO 0.4s cubic-bezier(0.78,0.13,0.16,0.87);
}

.c0 button {
  color: var(--px-alert-message-dismiss-icon-color,gray);
  outline: none;
  max-width: 50px;
}

.c0 button svg {
  height: 1.33333rem;
  width: 1.33333rem;
  fill: var(--px-alert-message-dismiss-icon-color,gray);
  stroke: var(--px-alert-message-dismiss-icon-color,gray);
}

.c2 {
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
  -webkit-flex: 2;
  -ms-flex: 2;
  flex: 2;
  margin-left: 0.66667rem;
  margin-right: 0.66667rem;
}

.c4 {
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
  position: relative;
}

.c4 .dismiss {
  width: 44px;
  height: 44px;
  min-width: auto;
}

.c1 {
  min-height: 4rem;
  width: 4rem;
  margin-left: -0.6666rem;
  margin-top: -0.6666rem;
  margin-bottom: -0.6666rem;
  padding-top: 1rem;
  background-color: var(--px-alert-message-color--information,blue);
}

.c3 {
  color: var(--px-alert-message-text-color,black);
  line-height: 1.06667rem;
  font-size: 0.8rem;
  padding: 0;
  margin: 0;
}

<div
  className="c0"
>
  <div
    className="c1"
    type="information"
  />
  <div
    className="c2"
  >
    <div
      className="c3"
    >
      <strong>
         
      </strong>
      <span />
      <div>
        This is a message
      </div>
    </div>
  </div>
  <div
    className="c4"
  />
</div>
`;

exports[`AlertMessage matches snapshot 1`] = `
.c0 {
  visibility: hidden;
  display: none;
  width: var(--px-alert-message-width,auto);
  margin: 0.33333rem;
  padding: 0.66667rem;
  min-height: 4rem;
  word-wrap: break-word;
  word-break: break-word;
  background-color: var(--px-alert-message-background-color,white);
  position: relative;
  min-height: 0px;
  opacity: 0;
  max-height: 0px;
  padding-top: 0px;
  padding-bottom: 0px;
  box-shadow: 0px 4px 8px var(--px-shadow-notification,rgba(0,0,0,0.2));
  -webkit-animation: LkNfV 0.4s cubic-bezier(0.78,0.13,0.16,0.87);
  animation: LkNfV 0.4s cubic-bezier(0.78,0.13,0.16,0.87);
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
  overflow: hidden;
  height: 4rem;
  visibility: visible;
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  opacity: 1;
  max-height: none;
  padding-top: 0.66667rem;
  padding-bottom: 0.66667rem;
  -webkit-animation: ayxbO 0.4s cubic-bezier(0.78,0.13,0.16,0.87);
  animation: ayxbO 0.4s cubic-bezier(0.78,0.13,0.16,0.87);
}

.c0 button {
  color: var(--px-alert-message-dismiss-icon-color,gray);
  outline: none;
  max-width: 50px;
}

.c0 button svg {
  height: 1.33333rem;
  width: 1.33333rem;
  fill: var(--px-alert-message-dismiss-icon-color,gray);
  stroke: var(--px-alert-message-dismiss-icon-color,gray);
}

.c2 {
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
  -webkit-flex: 2;
  -ms-flex: 2;
  flex: 2;
  margin-left: 0.66667rem;
  margin-right: 0.66667rem;
}

.c4 {
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
  position: relative;
}

.c4 .dismiss {
  width: 44px;
  height: 44px;
  min-width: auto;
}

.c1 {
  min-height: 4rem;
  width: 4rem;
  margin-left: -0.6666rem;
  margin-top: -0.6666rem;
  margin-bottom: -0.6666rem;
  padding-top: 1rem;
  background-color: var(--px-alert-message-color--important,red);
}

.c3 {
  color: var(--px-alert-message-text-color,black);
  line-height: 1.06667rem;
  font-size: 0.8rem;
  padding: 0;
  margin: 0;
}

<div
  className="c0"
>
  <div
    className="c1"
    type="important"
  />
  <div
    className="c2"
  >
    <div
      className="c3"
    >
      <strong>
         
      </strong>
      <span />
      <div />
    </div>
  </div>
  <div
    className="c4"
  />
</div>
`;

exports[`AlertMessage success matches snapshot 1`] = `
.c0 {
  visibility: hidden;
  display: none;
  width: var(--px-alert-message-width,auto);
  margin: 0.33333rem;
  padding: 0.66667rem;
  min-height: 4rem;
  word-wrap: break-word;
  word-break: break-word;
  background-color: var(--px-alert-message-background-color,white);
  position: relative;
  min-height: 0px;
  opacity: 0;
  max-height: 0px;
  padding-top: 0px;
  padding-bottom: 0px;
  box-shadow: 0px 4px 8px var(--px-shadow-notification,rgba(0,0,0,0.2));
  -webkit-animation: LkNfV 0.4s cubic-bezier(0.78,0.13,0.16,0.87);
  animation: LkNfV 0.4s cubic-bezier(0.78,0.13,0.16,0.87);
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
  overflow: hidden;
  height: 4rem;
  visibility: visible;
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  opacity: 1;
  max-height: none;
  padding-top: 0.66667rem;
  padding-bottom: 0.66667rem;
  -webkit-animation: ayxbO 0.4s cubic-bezier(0.78,0.13,0.16,0.87);
  animation: ayxbO 0.4s cubic-bezier(0.78,0.13,0.16,0.87);
}

.c0 button {
  color: var(--px-alert-message-dismiss-icon-color,gray);
  outline: none;
  max-width: 50px;
}

.c0 button svg {
  height: 1.33333rem;
  width: 1.33333rem;
  fill: var(--px-alert-message-dismiss-icon-color,gray);
  stroke: var(--px-alert-message-dismiss-icon-color,gray);
}

.c2 {
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
  -webkit-flex: 2;
  -ms-flex: 2;
  flex: 2;
  margin-left: 0.66667rem;
  margin-right: 0.66667rem;
}

.c4 {
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
  position: relative;
}

.c4 .dismiss {
  width: 44px;
  height: 44px;
  min-width: auto;
}

.c1 {
  min-height: 4rem;
  width: 4rem;
  margin-left: -0.6666rem;
  margin-top: -0.6666rem;
  margin-bottom: -0.6666rem;
  padding-top: 1rem;
  background-color: var(--px-alert-label-background-color--healthy,green);
}

.c3 {
  color: var(--px-alert-message-text-color,black);
  line-height: 1.06667rem;
  font-size: 0.8rem;
  padding: 0;
  margin: 0;
}

<div
  className="c0"
>
  <div
    className="c1"
    type="success"
  />
  <div
    className="c2"
  >
    <div
      className="c3"
    >
      <strong>
         
      </strong>
      <span />
      <div>
        This is a message
      </div>
    </div>
  </div>
  <div
    className="c4"
  />
</div>
`;

exports[`AlertMessage warning matches snapshot 1`] = `
.c0 {
  visibility: hidden;
  display: none;
  width: var(--px-alert-message-width,auto);
  margin: 0.33333rem;
  padding: 0.66667rem;
  min-height: 4rem;
  word-wrap: break-word;
  word-break: break-word;
  background-color: var(--px-alert-message-background-color,white);
  position: relative;
  min-height: 0px;
  opacity: 0;
  max-height: 0px;
  padding-top: 0px;
  padding-bottom: 0px;
  box-shadow: 0px 4px 8px var(--px-shadow-notification,rgba(0,0,0,0.2));
  -webkit-animation: LkNfV 0.4s cubic-bezier(0.78,0.13,0.16,0.87);
  animation: LkNfV 0.4s cubic-bezier(0.78,0.13,0.16,0.87);
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
  overflow: hidden;
  height: 4rem;
  visibility: visible;
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  opacity: 1;
  max-height: none;
  padding-top: 0.66667rem;
  padding-bottom: 0.66667rem;
  -webkit-animation: ayxbO 0.4s cubic-bezier(0.78,0.13,0.16,0.87);
  animation: ayxbO 0.4s cubic-bezier(0.78,0.13,0.16,0.87);
}

.c0 button {
  color: var(--px-alert-message-dismiss-icon-color,gray);
  outline: none;
  max-width: 50px;
}

.c0 button svg {
  height: 1.33333rem;
  width: 1.33333rem;
  fill: var(--px-alert-message-dismiss-icon-color,gray);
  stroke: var(--px-alert-message-dismiss-icon-color,gray);
}

.c2 {
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
  -webkit-flex: 2;
  -ms-flex: 2;
  flex: 2;
  margin-left: 0.66667rem;
  margin-right: 0.66667rem;
}

.c4 {
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
  position: relative;
}

.c4 .dismiss {
  width: 44px;
  height: 44px;
  min-width: auto;
}

.c1 {
  min-height: 4rem;
  width: 4rem;
  margin-left: -0.6666rem;
  margin-top: -0.6666rem;
  margin-bottom: -0.6666rem;
  padding-top: 1rem;
  background-color: var(--px-alert-message-color--warning,orange);
}

.c3 {
  color: var(--px-alert-message-text-color,black);
  line-height: 1.06667rem;
  font-size: 0.8rem;
  padding: 0;
  margin: 0;
}

<div
  className="c0"
>
  <div
    className="c1"
    type="warning"
  />
  <div
    className="c2"
  >
    <div
      className="c3"
    >
      <strong>
         
      </strong>
      <span />
      <div>
        This is a message
      </div>
    </div>
  </div>
  <div
    className="c4"
  />
</div>
`;
